### PR TITLE
refactor: update `blas/ext/base/sdsapxsum` to follow current project conventions

### LINKED ISSUE #1521: [RFC]: refactor `blas/ext/base/sdsapxsum` to follow current project conventions
### Description

This RFC proposes refactoring [`@stdlib/blas/ext/base/sdsapxsum`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/blas/ext/base/sdsapxsum) to follow current project conventions.

These conventions are outlined in more detail in https://github.com/stdlib-js/stdlib/issues/788. Namely, we want to migrate from C++ add-on interfaces to C add-on interfaces and then make various style and simplification changes.

Some example commits which perform most, if not all, of the requested changes include https://github.com/stdlib-js/stdlib/commit/d66d80cfcbab1da95cb5ae8cfae4a1fc4c5129d1 (saxpy),  https://github.com/stdlib-js/stdlib/commit/3c82fe7b6c0f9fd6ef02c86d6e2024d708ac8c14 (cswap), and https://github.com/stdlib-js/stdlib/commit/5b15c9675be434c0e11b1ec1e1f78d7b089abbcd (dasum).

### Related Issues

Related issues #788, #1152.

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by AuenKr**
Resolves #1521 .

## Description

> What is the purpose of this pull request?

This pull request:

-   update `@stdlib/blas/ext/base/sdsapxsum` to follow current project conventions

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #1521 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

File: lib/node_modules/@stdlib/blas/ext/base/sdsapxsum/docs/repl.txt [hunk 1]
```diff
@@ -92,8 +89,7 @@
 
     // Using offset parameter:
     > var x = new {{alias:@stdlib/array/float32}}( [ 1.0, -2.0, 3.0, 2.0, 5.0, -1.0 ] );
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/sdsapxsum/docs/repl.txt]: Already defined above (had a mistake already in `repl.txt`) Let's fix it:

```suggestion
    > x = new {{alias:@stdlib/array/float32}}( [ 1.0, -2.0, 3.0, 2.0, 5.0, -1.0 ] );
```

File: lib/node_modules/@stdlib/blas/ext/base/sdsapxsum/examples/index.js [hunk 2]
```diff
@@ -18,18 +18,11 @@
...
 var sdsapxsum = require( './../lib' );
 
-var x;
-var i;
-
-x = new Float32Array( 10 );
-for ( i = 0; i < x.length; i++ ) {
-	x[ i ] = round( randu()*100.0 );
-}
+var x = filledarrayBy( 10, 'float32', discreteUniform( 0, 100 ));
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/sdsapxsum/examples/index.js]:
```suggestion
var x = filledarrayBy( 10, 'float32', discreteUniform( 0, 100 ) );
```

File: lib/node_modules/@stdlib/blas/ext/base/sdsapxsum/src/addon.c [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+*/
+static napi_value addon( napi_env env, napi_callback_info info ) {
+	STDLIB_NAPI_ARGV( env, info, argv, argc, 4 );
+	STDLIB_NAPI_ARGV_INT64( env, N, argv, 0 );
+	STDLIB_NAPI_ARGV_FLOAT( env, N, argv, 1 );
+	STDLIB_NAPI_ARGV_INT64( env, stride, argv, 3 );
+	STDLIB_NAPI_ARGV_STRIDED_FLOAT32ARRAY( env, X, N, stride, argv, 2 );
+
+	napi_value v;
+	status = napi_create_double( env, stdlib_strided_sdsapxsum( N, alpha, X, stride ), &v );
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/sdsapxsum/src/addon.c]: Let's explicitly cast to double-precision:

```suggestion
	status = napi_create_double( env, (double)stdlib_strided_sdsapxsum( N, alpha, X, stride ), &v );
```

**Review by Planeshifter — CHANGES_REQUESTED** [review]
Looks good; just a few changes needed and we can merge your PR. Thanks!

**Comment by AuenKr** [comment]
@Planeshifter, applied all suggestions.

File: lib/node_modules/@stdlib/blas/ext/base/sdsapxsum/lib/ndarray.native.js [hunk 4]
```diff
@@ -38,20 +39,17 @@ var addon = require( './sdsapxsum.native.js' );
...
+* var v = sdsapxsum( 4, 5.0, x, 2, 1 );
 * // returns 25.0
 */
 function sdsapxsum( N, alpha, x, stride, offset ) {
 	var view;
-	if ( stride < 0 ) {
-		offset += (N-1) * stride;
-	}
-	view = new Float32Array( x.buffer, x.byteOffset+(x.BYTES_PER_ELEMENT*offset), x.length-offset ); // eslint-disable-line max-len
+
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/sdsapxsum/lib/ndarray.native.js]:
```suggestion
```

File: lib/node_modules/@stdlib/blas/ext/base/sdsapxsum/manifest.json [hunk 5]
```diff
@@ -1,42 +1,78 @@
...
+        "./src/sdsapxsum.c"
+      ],
+      "include": [
+        "./include"
+      ],
+      "libraries": [
+        "-lm"
+      ],
+      "libpath": [],
+      "dependencies": []
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/sdsapxsum/manifest.json]: You're missing the `sdsapxsumpw` dependency here and below.

**Review by kgryte — CHANGES_REQUESTED** [review]
Thanks for working on this, @AuenKr. Need to add the missing dependency in the `manifest.json` configuration objects, and this should be ready for merge.

**Comment by AuenKr** [comment]
@kgryte, added dependency in `manifest.json` file

**Review by Planeshifter — APPROVED** [review]
LGTM! Thanks.

**Pull request merged**